Utils/API/server/lib/dkb/api/storages/es.py

```diff
@@ -487,3 +490,51 @@ def task_kwsearch(**kwargs):
         task['output_dataset'] = [ds['_source'] for ds in datasets]
         result['_data'].append(task)
     return result
+
+
+def get_output_formats(tags):
+    formats = []
+    for tag in tags:
+        f = OUTPUT_FORMATS.get(tag)
+        if f:
+            formats += f
+    return formats
+
+
+def get_derivation_statistics_for_output(project, tags, output_format):
+    query = dict(TASK_KWARGS)
+    kwargs = {'project': project, 'ctag': tags, 'output': output_format}
+    query['body'] = get_query('deriv', **kwargs)
+    r = client().search(**query)
+    try:
+        total = r['hits']['total']
+        result_events = (r['aggregations']['output_datasets']['not_removed']
+                         ['format']['sum_events']['value'])
+        result_bytes = (r['aggregations']['output_datasets']['not_removed']
+                        ['format']['sum_bytes']['value'])
+        input_events = r['aggregations']['input_events']['value']
+        input_bytes = r['aggregations']['input_bytes']['value']
+        ratio = 0
+        if input_bytes != 0:
+            ratio = float(result_bytes) / float(input_bytes)
+        events_ratio = 0
+        if input_events != 0:
+            events_ratio = float(result_events) / float(input_events)
+    except Exception:
+        total = 0
+        ratio = 0
+        events_ratio = 0
+    return {'total': total, 'ratio': ratio, 'events_ratio': events_ratio}
+
+
+def task_derivation_statistics(**kwargs):
+    init()
+    project = kwargs.get('project').lower()
+    tags = [tag for tag in kwargs.get('amitag').split(',') if tag]
```

Comment:
Would be better to have `amitag`s passed as array initially: `...?amitag=TAG1&amitag=TAG2&...`; in this case `kwargs['amitag']` will have value `['TAG1', 'TAG2']`. It is not difficult to parse comma-separated string, of course, and AMI tags are not supposed to have comma within themselves -- but strict version is just more strict and leaves less space for mis-interpreting and mis-use. Also, it is the way keywords search is implemented (`...?kw=KEYWORD1&kw=KEYWORD2&...`), and I think we should keep same style of parameters handling in all methods.

Reply:
Good idea, done.